dist/subscribe.php

```diff
@@ -0,0 +1,90 @@
+<?php
+use PHPMailer\PHPMailer\PHPMailer;
+use PHPMailer\PHPMailer\SMTP;
+use PHPMailer\PHPMailer\Exception;
+/*
+ * Load PHP mailer class using composer.
+ */
+include "include/vendor/autoload.php";
+/*
+ * Regex pattern for email.
+ * Copyright Michael Rushton.
+ * Feel free to use and redistribute this code. But please keep this copyright notice.
+ * http://lxr.php.net/xref/PHP_5_4/ext/filter/logical_filters.c#501
+ */
+$pattern = '/^(?!(?:(?:\\x22?\\x5C[\\x00-\\x7E]\\x22?)|(?:\\x22?[^\\x5C\\x22]\\x22?)){255,})(?!(?:(?:\\x22?\\x5C[\\x00-\\x7E]\\x22?)|(?:\\x22?[^\\x5C\\x22]\\x22?)){65,}@)(?:(?:[\\x21\\x23-\\x27\\x2A\\x2B\\x2D\\x2F-\\x39\\x3D\\x3F\\x5E-\\x7E]+)|(?:\\x22(?:[\\x01-\\x08\\x0B\\x0C\\x0E-\\x1F\\x21\\x23-\\x5B\\x5D-\\x7F]|(?:\\x5C[\\x00-\\x7F]))*\\x22))(?:\\.(?:(?:[\\x21\\x23-\\x27\\x2A\\x2B\\x2D\\x2F-\\x39\\x3D\\x3F\\x5E-\\x7E]+)|(?:\\x22(?:[\\x01-\\x08\\x0B\\x0C\\x0E-\\x1F\\x21\\x23-\\x5B\\x5D-\\x7F]|(?:\\x5C[\\x00-\\x7F]))*\\x22)))*@(?:(?:(?!.*[^.]{64,})(?:(?:(?:xn--)?[a-z0-9]+(?:-+[a-z0-9]+)*\\.){1,126}){1,}(?:(?:[a-z][a-z0-9]*)|(?:(?:xn--)[a-z0-9]+))(?:-+[a-z0-9]+)*)|(?:\\[(?:(?:IPv6:(?:(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){7})|(?:(?!(?:.*[a-f0-9][:\\]]){7,})(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,5})?::(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,5})?)))|(?:(?:IPv6:(?:(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){5}:)|(?:(?!(?:.*[a-f0-9]:){5,})(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,3})?::(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,3}:)?)))?(?:(?:25[0-5])|(?:2[0-4][0-9])|(?:1[0-9]{2})|(?:[1-9]?[0-9]))(?:\\.(?:(?:25[0-5])|(?:2[0-4][0-9])|(?:1[0-9]{2})|(?:[1-9]?[0-9]))){3}))\\]))$/iD';
+/*
+ * Constants for email credentials.
+ * Makes it easy to edit the username, password, port, etc.
+ * LOAD from a env file that's in the root folder, the template can be found in
+ * .env.example file
+ */
+$dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
+$dotenv->load();
+$constants = array(
+  "HOST" => $_ENV['HOST'],
+  "USERNAME" => $_ENV['USERNAME'],
+  "PASSWORD" => $_ENV['PASSWORD'],
+  "PORT" => (int)$_ENV['PORT'],
+  "FROM" => $_ENV['FROM']
+);
+
+if (!isset($_POST["mail_input"])) {
+  header("location: ./");
+  exit();
+}
+$mail = new PHPMailer(true);
+$emailString = $_POST["mail_input"];
+/*
+ * Setup a response string for returning a response, in case the email was bad.
+ */
+$responseString = "";
+/*
+ * Server side email validation.
+ */
+if (preg_match($pattern, $emailString) !== 1) {
+  $responseString = "Please enter an email";
+  echo $responseString;
+  exit();
+}
+try {
+  /*
+   * Server settings.
+  */
+  $mail->SMTPDebug = SMTP::DEBUG_SERVER;
+  $mail->isSMTP();
+  $mail->Host       = $constants["HOST"];
+  $mail->SMTPAuth   = true;
+  $mail->Username   = $constants["USERNAME"];
+  $mail->Password   = $constants["PASSWORD"];
+  $mail->SMTPSecure = PHPMailer::ENCRYPTION_STARTTLS;
+  $mail->Port       = $constants["PORT"];
+  $mail->SMTPSecure = 'ssl';
+  /*
+   * Should be turned on in production.
+   * $mail->SMTPDebug  = false;
+   */
```

Comment:
This should be dependent on an .env constant, such as for example "DEBUG = true/false".

Reply:
I don't see why it should be. The .env file is for sensitive configurable settings. This doesn't need an env entry, one doesn't need to create a .env just for turning debugging on and off

Reply:
I see it like this: .env is short for ENVironment. It's a configuration file. When I want to change something about the environment (includes IMO switching between production and development), I should be able to change this in the config file and not by editing source code.

Deployers can assume, that the .env file will almost never change (and if, it should be written in large letters on the PR description -- note for future PRs 😃), whereas the actual source code can change with any update. This would result in this change possibly being overwritten (because this code is tracked by git).